### DISCUSSION TITLE
Hook into Larastan's schema analysis

### DISCUSSION THREAD

**Discussion by rudiedirkx** [larastan/larastan#1926] [Ideas]
It would be nice to hook into Larastan's extensive existing schema analysis. My app has a few custom Blueprint methods that create and drop columns, eg `Blueprint::statusChanged()` creates `status_changed_at` and `status_changed_by`. I can copy the whole Larastan schema code, OR Larastan could give me a hook into its. Probably in `SchemaAggregator`.

**Comment by rudiedirkx** [comment]
I also override `Blueprint::timestamp()` to make an `int` column instead of `timestamp`, so it shouldn't parse as `string` but `int`. Maybe that's too hacky too support, but a hook into Larastan could also 'fix' that.

**Reply by rudiedirkx** [reply]
Also solved by the schema scan 👍

**Comment by calebdw** [comment]
You can also just dump the migrations and then have Larastan scan the dumps which should simplify things (unless you're using Postgres)

**Reply by rudiedirkx** [reply]
Waat!! Is that a thing?? Larastan scans the `.sql` file? That's brilliant, I want that, how do I do that? 😆

**Reply by calebdw** [reply]
It's on by default lol: https://github.com/larastan/larastan/blob/2.x/docs/custom-config-parameters.md#squashedmigrationspath

Larastan first scans any dumps and then it will scan the migrations. Right now it only supports a single connection, but #1879 adds support for multiple connections

**Reply by rudiedirkx** [reply]
I love it! Like you said, much easier than hooks and code ❤️
